apisix/plugins/csrf.lua

```diff
@@ -0,0 +1,169 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+local core = require("apisix.core")
+local ngx = ngx
+local plugin_name = "csrf"
+local ngx_encode_base64 = ngx.encode_base64
+local ngx_decode_base64 = ngx.decode_base64
+local request_method = ngx.var.request_method
+local timer = ngx.time
+local cookie_time = ngx.cookie_time
+local ck = require "resty.cookie"
+local math = math
+
+local schema = {
+    type = "object",
+    properties = {
+      key = {
```

Comment:
From the way this variable is used, it seems more appropriate to use `hash_salt`

Reply:
It is used to encrypt the token to prevent it from being forged.

Reply:
Got it. Thanks for reply.